field_utils/FieldCondensedArray.F90

```diff
@@ -11,6 +11,7 @@ module mapl3g_FieldCondensedArray
    implicit none
    private
    public :: assign_fptr_condensed_array
+   public :: get_fptr_shape
```

Comment:
The intent is that the vertical regridder use the higher level procedure that simply returns a pointer to the correctly shaped 3D array.

We will need to understand why this does not work in your use case.

Reply:
This doesn't need to be public when using the higher level function `assign_fptr_condensed_array`. I'll make the change.